File: app/src/main/java/co/tpcreative/supersafe/common/entities/BreakInAlertsEntity.java
package co.tpcreative.supersafe.common.entities;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;
import java.io.Serializable;
import co.tpcreative.supersafe.model.BreakInAlertsEntityModel;

@Entity(tableName = "breakinalerts")
public class BreakInAlertsEntity implements Serializable{
    @PrimaryKey(autoGenerate = true)
    public int id;
    public String fileName;
    public long time;
    public String pin;

    @Ignore
    private static BreakInAlertsEntity instance;

    @Ignore
    public static BreakInAlertsEntity getInstance(){
        if (instance==null){
            instance = new BreakInAlertsEntity();
        }
        return instance;
    }

    @Ignore
    public BreakInAlertsEntity(){
        this.fileName = null;
        this.time = 0;
        this.pin = null;
    }

    @Ignore
    public BreakInAlertsEntity(BreakInAlertsEntityModel value){
        this.id = value.id;
        this.fileName = value.fileName;
        this.time = value.time;
        this.pin = value.pin;
    }

    public BreakInAlertsEntity(int id, String fileName,long time,String pin){
        this.id = id;
        this.fileName = fileName;
        this.time = time;
        this.pin = pin;
    }
}

File: app/src/main/java/co/tpcreative/supersafe/common/entities/BreakInAlertsEntity.kt
package co.tpcreative.supersafe.common.entities
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import co.tpcreative.supersafe.model.BreakInAlertsEntityModel
import java.io.Serializable

@Entity(tableName = "breakinalerts")
class BreakInAlertsEntity : Serializable {
    @PrimaryKey(autoGenerate = true)
    var id = 0
    var fileName: String?
    var time: Long
    var pin: String?

    @Ignore
    constructor() {
        fileName = null
        time = 0
        pin = null
    }

    @Ignore
    constructor(value: BreakInAlertsEntityModel) {
        id = value?.id
        fileName = value?.fileName
        time = value?.time
        pin = value?.pin
    }

    constructor(id: Int, fileName: String?, time: Long, pin: String?) {
        this.id = id
        this.fileName = fileName
        this.time = time
        this.pin = pin
    }

    companion object {
        @Ignore
        private var instance: BreakInAlertsEntity? = null

        @Ignore
        fun getInstance(): BreakInAlertsEntity? {
            if (instance == null) {
                instance = BreakInAlertsEntity()
            }
            return instance
        }
    }
}